### ISSUE TITLE
Infinte loop when cleaning up connection; get_conn hangs forever

### ISSUE THREAD

**Issue by nemosupremo** [blackbeam/mysql_async#126]
I'm trying to debug an issue - I have a pool with a constraint set to a minimum of 0 and a max of 1.  After exhausting my connections `get_conn` hangs and my program enters an infinite loop. Using gdb I was able to find where the hang occurs, and it happens in `cleanup`.

https://github.com/blackbeam/mysql_async/blob/be891f5296019b699124224137dcac7f5032bdca/src/conn/mod.rs#L751

For some reason `pending_result` is *always* `Some`, so `self.drop_result().await?;` is called, but then on the next iteration of the loop, `pending_result` is still Some.

It looks like `ResultSet` is of `ResultSetMeta::Text`, but it never seems to actually get dropped

**Comment by nemosupremo** [comment]
The issue doesn't occur with `=0.24.0`

**Comment by blackbeam** [comment]
Seems like #123 introduced a regression. Thanks for report

**Comment by blackbeam** [comment]
Couldn't reproducing it 🤔

Tried:

* `Text([])` pending result
* `Text([..])` pending result
* `Binary([])` pending result
* `Binary([..])` pending result
* `Text` multi-result set with empty and non-empty sets
* all of above within a transaction

Could you please share some code?

**Comment by nemosupremo** [comment]
Yes, I was able to put together an example - it was pretty late when I made this PR, so the bug is kind of sloppy. Here is how to reproduce the error on the latest version:

```rust
use mysql_async::prelude::*;

#[tokio::main]
async fn main() {

    let pool = mysql_async::Pool::new("mysql://root:password@127.0.0.1/sample?pool_min=0&pool_max=1&prefer_socket=false");
    let mut conn = pool.get_conn().await.expect("failed to get connection");
    let q = "
        CREATE TABLE IF NOT EXISTS `sample_table` (
            `key` SERIAL PRIMARY KEY,
            `value` INT NOT NULL
        ) engine=InnoDB CHARSET=utf8mb4;

        DROP TRIGGER IF EXISTS `sample_trigger`;

        CREATE TRIGGER `sample_trigger` AFTER INSERT ON `sample_table`
        FOR EACH ROW
        BEGIN
            op.value = op.value + 1
        END;

        CREATE TRIGGER `sample_trigger` AFTER INSERT ON `sample_table`
        FOR EACH ROW
        BEGIN
            op.value = op.value + 1
        END;
    ";
    match conn.query_drop(q).await {
        Ok(_) => println!("The query succeeded when it should have failed!, the next conn.get will fail"),
        Err(err) => panic!("Query failed successfully: {}", err),
    };
    drop(conn);
    // this will hang
    let mut conn = pool.get_conn().await.expect("couldn't get connection");
    println!("Done");
}
```

Basically the first query will try to execute a query that will try to create multiple triggers. This query _should_ fail, but doesnt, and the underyling connection is stuck in limbo. The next call to `get_conn` fails and hangs forever.

**Comment by blackbeam** [comment]
Fix published in [v0.24.2](https://github.com/blackbeam/mysql_async/releases/tag/v0.24.2)
